Importer/CsvImporter.php

```diff
@@ -248,16 +241,18 @@ protected function checkHeadersAreValid($csv)
     }
 
     /**
-     * @param ImporterErrorInterface $error
+     * @param DBALException $exception
+     *
+     * @return string
      */
-    protected function handleImporterError(ImporterErrorInterface $error)
+    private function resolveDBALException(DBALException $exception)
```

Comment:
True, but this part is not the only part in the `CsvImporter` that is not relevant to the specific `Csv`. As we discussed while implementing this `CsvImporter`, when we get a second usage of `Importer`, then we can have an `AbstractImporter` and move this logic there.

If for whatever reason still we want to create an `AbstractImporter` and simplify the `CsvImporter`, then I would suggest to do this change on a further refactoring.